service/src/main/kotlin/fi/espoo/evaka/reports/ServiceVoucherValueReport.kt

```diff
@@ -100,24 +113,35 @@ data class ServiceVoucherValueRow(
     val childFirstName: String,
     val childLastName: String,
     val childDateOfBirth: LocalDate,
+    val childGroupName: String?,
     val unitId: UUID,
     val unitName: String,
     val areaId: UUID,
     val areaName: String,
     val serviceVoucherPeriod: Period,
     val serviceVoucherValue: Int,
     val serviceVoucherCoPayment: Int,
-    val serviceVoucherServiceCoefficient: Int
+    val derivatives: ValueRowDerivatives? = null
+)
+
+data class ValueRowDerivatives(
+    val realizedAmount: Int,
+    val coefficient: BigDecimal,
+    val realizedPeriod: ClosedPeriod,
+    val numberOfDays: Long
 )
 
-private fun realizedMonthlyAmount(value: Int, decisionPeriod: Period, monthPeriod: ClosedPeriod): Int {
+private fun realizedMonthlyAmount(value: Int, decisionPeriod: Period, monthPeriod: ClosedPeriod): ValueRowDerivatives {
     val realizedPeriod = ClosedPeriod(
         maxOf(decisionPeriod.start, monthPeriod.start),
         minOf(decisionPeriod.end ?: monthPeriod.end, monthPeriod.end)
     )
-    val coefficient = BigDecimal(realizedPeriod.durationInDays())
-        .divide(BigDecimal(monthPeriod.durationInDays()), 10, RoundingMode.HALF_UP)
-    return (BigDecimal(value) * coefficient).setScale(0, RoundingMode.HALF_UP).toInt()
+
+    val numberOfDays = realizedPeriod.durationInDays()
+    val coefficient = BigDecimal(numberOfDays).divide(BigDecimal(monthPeriod.durationInDays()), 10, RoundingMode.HALF_UP)
+    val realizedAmount = (BigDecimal(value) * coefficient).setScale(0, RoundingMode.HALF_UP).toInt()
+
+    return ValueRowDerivatives(realizedAmount, coefficient, realizedPeriod, numberOfDays)
```

Comment:
This `coefficient` is actually not the coefficient shown in the table. That coefficient is (probably) the combination of `age_coefficient` and `service_coefficient` on `voucher_value_decision_part`.

Reply:
Right, thanks... I'll use service_coefficient for now.